pinot-broker/src/main/java/org/apache/pinot/broker/requesthandler/BaseBrokerRequestHandler.java

```diff
@@ -428,6 +437,38 @@ public BrokerResponse handleRequest(JsonNode request, @Nullable RequesterIdentit
       _brokerMetrics.addMeteredTableValue(rawTableName, BrokerMeter.BROKER_RESPONSES_WITH_NUM_GROUPS_LIMIT_REACHED, 1);
     }
 
+    logBrokerResponse(requestStatistics, requestId, query, compilationStartTimeNs, brokerRequest,
+        numUnavailableSegments, serverStats, brokerResponse, executionEndTimeNs);
+    return brokerResponse;
+  }
+
+  /**
+   * Given a {@link BrokerRequest}, this function will determine if we can return a response without server-side query
+   * evaluation. This usually happens when the optimizer determines that the entire WHERE clause evaluates to false.
+   */
+  private boolean isResponsePossible(BrokerRequest brokerRequest) {
+    if (brokerRequest == null) { return true;}
+
+    Expression filterExpression = brokerRequest.getPinotQuery().getFilterExpression();
+    if (filterExpression != null && filterExpression.getType().equals(ExpressionType.LITERAL) && filterExpression
```

Comment:
We can add constants for literal `true` and `false` expression, and check if the filter expression equals to them for better readability

Reply:
Reworked this a bit by using TRUE/FALSE Boolean literal constants.